### PR TITLE
dev/reference/*: add `use sql statements`

### PR DESCRIPTION
**Pull request by huangxiuyan**
Via: https://github.com/pingcap/docs-cn/pull/1365
@lilin90 @WangXiangUSTC, PTAL, thanks!

### CODE REVIEW

File: dev/reference/tools/tidb-binlog/operation.md [hunk 1]
```diff
@@ -48,10 +48,11 @@ For how to pause, close, check, and modify the state of Drainer, see the [binlog
 Your distribution of TiDB or TiDB-Binlog may already include binlogctl. If not, download `binlogctl`:
 
 ```bash
-wget https://download.pingcap.org/binlogctl-new-linux-amd64.{tar.gz,sha256}
+wget https://download.pingcap.org/tidb-{version}-linux-amd64.tar.gz
+wget https://download.pingcap.org/tidb-{version}-linux-amd64.sha256
 
 # Check the file integrity. It should return OK.
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
# Check the file integrity. If the result is OK, the file is correct.
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 2]
```diff
@@ -127,3 +128,61 @@ Command example:
     ```
 
     This command generates a `{data-dir}/savepoint` file. This file stores the `tso` information which is needed for the initial start of Drainer.
+
+## Use SQL statements to control Pump/Drainer
+
+Execute the corresponded SQL statements in the TiDB server to check or modify the state of binlog.
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
To view or modify binlog related states, execute corresponding SQL statements in TiDB.
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 3]
```diff
@@ -127,3 +128,61 @@ Command example:
...
+
+- Check whether the binlog is enabled:
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
- Check whether binlog is enabled:
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 4]
```diff
@@ -127,3 +128,61 @@ Command example:
...
+    +---------------+-------+
+    | Variable_name | Value |
+    +---------------+-------+
+    | log_bin       |  ON   |
+    +---------------+-------+
+    ```
+
+    When the Value is `ON`, it means that the binlog is enabled.
+
+- Check the status of all the Pumps and Drainers:
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
- Check the status of all the Pump or Drainer nodes:
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 5]
```diff
@@ -127,3 +128,61 @@ Command example:
...
+    +----------|----------------|--------|--------------------|---------------------|
+    |  NodeID  |     Address    | State  |   Max_Commit_Ts    |    Update_Time      |
+    +----------|----------------|--------|--------------------|---------------------|
+    | drainer1 | 127.0.0.3:8249 | Online | 408553768673342532 | 2019-05-01 00:00:03 |
+    +----------|----------------|--------|--------------------|---------------------|
+    | drainer2 | 127.0.0.4:8249 | Online | 408553768673345531 | 2019-05-01 00:00:04 |
+    +----------|----------------|--------|--------------------|---------------------|
+    ```
+
+- Modify the state of the Pump/Drainer node:
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
- Modify the state of a Pump or Drainer node:
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 6]
```diff
@@ -127,3 +128,61 @@ Command example:
...
+    ```
+
+    ```bach
+    mysql> change drainer to node_state ='paused' for node_id 'drainer1'";
+    Query OK, 0 rows affected (0.01 sec)
+    ```
+
+> **Note:**
+>
+> - TiDB v2.1.7 or higher supports checking whether the binlog is enabled and the running status of Pump/Drainer.
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
> - Checking whether binlog is enabled and the running status of Pump or Drainer is supported in TiDB v2.1.7 and later versions.
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 7]
```diff
@@ -127,3 +128,61 @@ Command example:
...
+> - TiDB v3.0.0-rc.1 or higher supports modifying the state of Pump/Drainer. Only the state of the Pump/Drainer which is stored in Placement Driver (PD) can be changed. If you want to pause or close the node, the `binlogctl` is still required to use.
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
> - Modifying the status of Pump or Drainer is supported in TiDB v3.0.0-rc.1 and later versions. This feature only supports modifying the status of Pump or Drainer nodes stored in the Placement Driver (PD). To pause or close the node, use the `binlogctl` tool.
```

**Comment by huangxiuyan** [comment]
@lilin90 @WangXiangUSTC PTAL, thanks!

**Comment by huangxiuyan** [comment]
@lilin90 Correct the slash format, PTAL, thanks!

File: dev/reference/tools/tidb-binlog/operation.md [hunk 8]
```diff
@@ -127,3 +128,74 @@ Command example:
...
+    ```
+
+    ```bach
+    mysql> change drainer to node_state ='paused' for node_id 'drainer1'";
+    Query OK, 0 rows affected (0.01 sec)
+    ```
+
+> **Note:**
+>
+<<<<<<< HEAD
```
Comment (morgo) [inline dev/reference/tools/tidb-binlog/operation.md]: It looks like you are accidentally committing a git merge here.

Comment (huangxiuyan) [reply dev/reference/tools/tidb-binlog/operation.md]: It seems the conflict arises when I commit the suggestion on the github web while some commits are still in the desktop. Thanks for your reminder!

File: dev/reference/tools/tidb-binlog/operation.md [hunk 9]
```diff
@@ -7,51 +7,52 @@ aliases: ['/docs/tools/binlog/operation/']
...
-* `offline`: becomes offline. After Pump sents all the binlog data that it saves to Drainer, its state is switched to `offline`.
+* `pausing`: in the pausing process. It turns into this state after you use `kill` or press Ctrl + C to exit from the process. When Pump or Drainer exits all internal threads in safe, it becomes `paused`.
+* `paused`: has been stopped. While Pump is in this state, it rejects the request of writing binlog into it and does not provide the binlog for Drainer any more. When Drainer is in this state, it does not replicate data to the downstream. After Pump and Drainer exit normally from all the threads, they switch the state to `paused` and then exits from the process.
+* `closing`: in the offline process. The `binlogctl` is used to get Pump or Drainer offline and Pump or Drainer is in this state before the process exits. In this state, Pump does not accept new requests of writing binlog into it and waits for all the binlog data to be used up by Drainer.
+* `offline`: becomes offline. After Pump sent all the binlog data that it saves to Drainer, its state is switched to `offline`.
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
* `offline`: becomes offline. After Pump sends all the binlog data that it saves to Drainer, its state is switched to `offline`.
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 10]
```diff
@@ -7,51 +7,52 @@ aliases: ['/docs/tools/binlog/operation/']
...
 > * When Pump is `closing`, you need to guarantee that all the data has been consumed by all the Drainers that are not `offline`. So before making Pump offline, you need to guarantee all the Drainers are `online`. Otherwise, Pump cannot get offline normally.
 > * The binlog data that Pump saves is processed by GC only when it has been consumed by all the Drainers that are not `offline`.
 > * Close Drainer only when it will not be used any more.
 
-For how to pause, close, check, and modify the state of Drainer, see the [binlogctl guide](#binlogctl-guide) as follows.
+For how to pause, close, check, and modify the states of Drainer, see the [binlogctl guide](#binlogctl-guide) as follows.
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
For how to pause, close, check, and modify the state of Drainer, see the [binlogctl guide](#binlogctl-guide) as follows.
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 11]
```diff
@@ -7,51 +7,52 @@ aliases: ['/docs/tools/binlog/operation/']
...
 
 ## `binlogctl` guide
 
 [`binlogctl`](https://github.com/pingcap/tidb-tools/tree/master/tidb-binlog/binlogctl) is an operations tool for TiDB-Binlog with the following features:
 
 * Obtaining the current `tso` of TiDB cluster
-* Checking the Pump/Drainer state
-* Modifying the Pump/Drainer state
-* Pausing or closing Pump/Drainer
+* Checking the states of Pump or Drainer
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
* Checking the state of Pump or Drainer
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 12]
```diff
@@ -7,51 +7,52 @@ aliases: ['/docs/tools/binlog/operation/']
...
+* Modifying the states of Pump or Drainer
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
* Modifying the state of Pump or Drainer
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 13]
```diff
@@ -7,51 +7,52 @@ aliases: ['/docs/tools/binlog/operation/']
...
+* Pausing or closing Pump or Drainer
 
 ### Usage scenarios of `binlogctl`
 
 * It is the first time you run Drainer and you need to obtain the current `tso` of TiDB cluster.
-* When Pump/Drainer exits abnormally, its state is not updated and the service is affected. You can use this tool to modify the state.
-* An error occurs during synchronization and you need to check the running status and the Pump/Drainer state.
-* While maintaining the cluster, you need to pause or close Pump/Drainer.
+* When Pump or Drainer exits abnormally, its state is not updated and the service is affected. You can use this tool to modify the state.
+* An error occurs during data replication and you need to check the running status and states of Pump or Drainer.
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
* An error occurs during data replication and you need to check the running status and state of Pump or Drainer.
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 14]
```diff
@@ -81,39 +82,39 @@ Usage of binlogctl:
...
+    Set `cmd` as `pumps` or `drainers` to check the state of all the Pump or Drainer nodes. For example,
 
     ```bash
     bin/binlogctl -pd-urls=http://127.0.0.1:2379 -cmd pumps
 
     INFO[0000] pump: {NodeID: ip-172-16-30-67:8250, Addr: 172.16.30.192:8250, State: online, MaxCommitTS: 405197570529820673, UpdateTime: 2018-12-25 14:23:37 +0800 CST}
     ```
 
-- Modify the Pump/Drainer state:
+- Modify the states of Pump or Drainer:
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]: Please see my previous suggestion.

```suggestion
- Modify the state of Pump or Drainer:
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 15]
```diff
@@ -81,39 +82,39 @@ Usage of binlogctl:
...
 
-    Set `cmd` as `update-pump` or `update-drainer` to modify the state of Pump or Drainer, which can be `online`, `pausing`, `paused`, `closing` or `offline`. 
+    Set `cmd` as `update-pump` or `update-drainer` to modify the states of Pump or Drainer, which can be `online`, `pausing`, `paused`, `closing` or `offline`.
 
     ```bash
     bin/binlogctl -pd-urls=http://127.0.0.1:2379 -cmd update-pump -node-id ip-127-0-0-1:8250 -state paused
     ```
 
-    This command modifies the Pump/Drainer state saved in PD.
+    This command modifies the states of Pump or Drainer saved in the Placement Driver (PD).
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
    This command modifies the state of Pump or Drainer saved in the Placement Driver (PD).
```

File: dev/reference/tools/tidb-binlog/operation.md [hunk 16]
```diff
@@ -127,3 +128,61 @@ Command example:
     ```
 
     This command generates a `{data-dir}/savepoint` file. This file stores the `tso` information which is needed for the initial start of Drainer.
+
+## Use SQL statements to control Pump or Drainer
```
Comment (lilin90) [inline dev/reference/tools/tidb-binlog/operation.md]:
```suggestion
## Use SQL statements to manage Pump or Drainer
```

**Comment by huangxiuyan** [comment]
@lilin90 PTAL, thanks!

**Comment by lilin90** [comment]
@WangXiangUSTC PTAL

**Comment by WangXiangUSTC** [comment]
LGTM

**Pull request merged**